lib/charms/prometheus/v1/prometheus.py

```diff
@@ -94,8 +93,7 @@ def remove_endpoint(self, address, port=80, rel_id=None):
                 :class:`PrometheusConsumer` has been instantiated in
                 `multi` mode.
         """
-        if rel_id is None:
-            rel_id = self.relation_id
+        rel_id = self.framework.model.get_relation(self._relation_name, rel_id).id
```

Comment:
reminds me the ren & stimpy episode where stimpy climbed into his own belly button

Reply:
What are you trying to achieve here ? Please explain.  I do not see a commit message or a comment. Are you trying to fix a bug for instance ? What is the bug ?